src/00b3_survey_mics/script.R

```diff
@@ -211,146 +290,183 @@ mics_final <- circ_recoded %>%
+      ## Note: keeping Mzuzu City as part of Mzimba district
+      survey_id == "MWI2019MICS" & grepl("Blantyre|Lilongwe|Zomba", area_name) ~ sub("Urban", "City", area_name) %>%
+        sub(" +Rural", "", .),
+      survey_id == "MWI2019MICS"  ~ sub(" Urban| Rural", "", area_name) %>%
+        recode("Nkhata Bay" = "Nkhatabay"),
       TRUE ~ area_name
     )
-  ) %>% 
-  select(-c(dob, doi)) %>% 
-  # rename weights column to match circumcision colnames
-  rename(indweight = mnweight)
+  )
+
+
 
 # check that any survey ages are greater than circumcision ages
 # something wrong with circ_age! 8 times where circ_age > surveyed age
-# Let Oli know 
-n <- mics_final %>% 
-  filter(circ_age > age) %>% 
+# Let Oli know
+n <- mics_final %>%
+  filter(circ_age > age) %>%
   nrow()
 
 print(n)
 
 # if there are only a few surveys where this is the case, can be safely dropped
 if (n > 0 && n < 20) {
- mics_final <- mics_final %>% 
-   filter(is.na(circ_age) | is.na(age) | circ_age <= age)
+  mics_final <- mics_final %>%
+    filter(is.na(circ_age) | is.na(age) | circ_age <= age)
+} else {
+  stop("Too many errors in circ_age > age; something likely wrong")
 }
 
 # add area_id to dataset
-areas_join <- areas %>% 
+areas_join <- areas %>%
   # # hh7 (i.e. area_name) maps to area lev 1, filter to avoid duplication
   filter(
-    (area_level == 1) |
-    # special cases
-    (iso3 == "MWI" & area_level %in% c(3, 5)) |  # need lev 3 for Mzimba
-    (iso3 == "NGA" & area_level == 2)
-  ) %>% 
-  select(area_id, area_name, area_level) %>% 
-  arrange() 
+    (area_level == 1 & !iso3 %in% c("MWI", "NGA")) |
+      # special cases
+      (iso3 == "MWI" & (area_name == "Mzimba" & area_level == 3 | area_level == 5)) |  # need lev 3 for Mzimba
+      (iso3 == "NGA" & area_level == 2)
+  ) %>%
+  select(iso3, area_id, area_name, area_level) %>%
+  arrange()
 
 # remove possible duplicate, lower level areas from areas_join (for MWI)
-duplicate_areas <- areas_join %>% 
-  group_by(area_name) %>% 
+duplicate_areas <- areas_join %>%
+  group_by(iso3, area_name) %>%
   summarise(
-    occurences = n(), 
-    area_id = area_id, 
+    occurences = n(),
+    iso3 = iso3,
+    area_id = area_id,
     area_level = area_level,
     max_area_level = max(area_level),
     .groups = "drop"
-  ) %>% 
-  filter(occurences > 1, area_level != max_area_level) %>% 
-  select(area_name, area_id)
-
-areas_join <- areas_join %>% 
-  anti_join(duplicate_areas, by = c("area_id", "area_name")) %>% 
-  select(area_id, area_name)
-  
-# split by iso3 before joining with areas_join, again to avoid duplication 
-mics_final_list <- split(mics_final, mics_final$iso3)
-mics_final <- bind_rows(lapply(seq_along(mics_final_list), function(i) {
-  mics_final_list[[i]] %>% 
-    left_join(
-      filter(areas_join, grepl(names(mics_final_list)[i], area_id)), 
-      by = "area_name"
-    )
-}))
+  ) %>%
+  filter(occurences > 1, area_level != max_area_level) %>%
+  select(iso3, area_name, area_id)
```

Comment:
A couple of comments on this:

* I think it's better to exactly pick the areas you want (in the edited line 420-423), rather than select multiple levels and the remove duplicately named.
* The previous code as written didn't quite work because it removed duplicates based on `area_name` being unique. This ended up dropping `Plateau` area for Benin (because that name also appeared in another country).   Solution was to add `iso3` to the joins to make unique within a country. Adding `iso3` to the joins also means that you don't need to do splitting by country and concatenating again.

Reply:
Why did you group by iso3 and then also have `summarise(iso3 = iso3)?

Reply:
Good catch. An error -- that is not needed.

Reply:
This is addressed in e0fcc21